File: core_common/src/main/java/com/huyingbao/core/common/dialog/CommonLoadingDialog.java
package com.huyingbao.core.common.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.huyingbao.core.arch.action.RxActionManager;
import com.huyingbao.core.arch.model.RxLoading;
import com.huyingbao.core.base.view.BaseDialogFragment;
import com.huyingbao.core.common.R;
import com.huyingbao.core.common.R2;
import com.huyingbao.core.util.CommonUtils;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.OnClick;
import dagger.Lazy;

/**
 * 进度提示
 * <p>
 * Created by liujunfeng on 2019/1/1.
 */
@Singleton
public class CommonLoadingDialog extends BaseDialogFragment implements DialogInterface.OnShowListener {
    @Inject
    Lazy<RxActionManager> mRxActionManagerLazy;

    @BindView(R2.id.progress_bar_loading)
    ProgressBar mProgressBarLoading;
    @BindView(R2.id.tv_loading_notice)
    TextView mTvLoadingNotice;
    @BindView(R2.id.tv_loading_cancel)
    TextView mTvLoadingCancel;

    private CharSequence message;
    private int messageInt;
    private RxLoading mRxLoading;

    public static CommonLoadingDialog newInstance() {
        return new CommonLoadingDialog();
    }

    @Inject
    public CommonLoadingDialog() {

    }

    @Override
    public int getLayoutId() {
        return R.layout.common_dialog_loading;
    }

    @Override
    public void afterCreate(@Nullable Bundle savedInstanceState) {

    }

    @Override
    public void onStart() {
        super.onStart();
        Dialog dialog = getDialog();
        if (dialog == null) {
            return;
        }
        Window window = dialog.getWindow();
        WindowManager.LayoutParams layoutParams = window.getAttributes();
        layoutParams.width = getResources().getDimensionPixelSize(R.dimen.dp_120);
        window.setAttributes(layoutParams);
    }

    @Override
    public void onShow(DialogInterface dialog) {
        if (message != null && message.length() > 0) {
            mTvLoadingNotice.setText(message);
            mTvLoadingNotice.setVisibility(View.VISIBLE);
        } else if (messageInt > 0) {
            mTvLoadingNotice.setText(messageInt);
            mTvLoadingNotice.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);
    }

    @Override
    public void onCancel(@NonNull DialogInterface dialog) {
        super.onCancel(dialog);
    }

    public void setMessage(CharSequence message) {
        this.message = message;
    }

    public void setRxLoading(RxLoading rxLoading) {
        mRxLoading = rxLoading;
    }

    /**
     * 取消所有注册的订阅Action
     */
    @OnClick(R2.id.tv_loading_cancel)
    public void cancel() {
        mRxActionManagerLazy.get().clear();
        CommonUtils.showShortToast(getContext(), "取消操作！");
        dismiss();
    }
}

File: module-common/src/main/java/com/huyingbao/module/common/ui/loading/CommonLoadingDialog.kt
package com.huyingbao.module.common.ui.loading

import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import androidx.annotation.StringRes
import com.huyingbao.core.base.common.dialog.BaseCommonDialog
import com.huyingbao.module.common.R
import kotlinx.android.synthetic.main.common_dialog_loading.*

/**
 * 进度提示
 *
 * Created by liujunfeng on 2019/1/1.
 */
class CommonLoadingDialog : BaseCommonDialog() {
    var clickListener: CommonLoadingDialogClickListener? = null

    @StringRes
    private var messageInt: Int = 0

    companion object {
        fun newInstance() = CommonLoadingDialog()
    }

    override fun getLayoutId() = R.layout.common_dialog_loading

    override fun afterCreate(savedInstanceState: Bundle?) {
        dialog?.setCanceledOnTouchOutside(false)
//        if (messageInt != 0) {
//            val message = getString(messageInt)
//            if (!TextUtils.isEmpty(message)) {
//                binding.tvLoadingNotice.text = message
//                binding.tvLoadingNotice.visibility = View.VISIBLE
//            }
//        }
//        binding.tvLoadingCancel.setOnClickListener {
//            clickListener?.onCancel()
//            dismiss()
//        }
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        messageInt = 0
        clickListener = null
    }

    override fun onCancel(dialog: DialogInterface) {
        super.onCancel(dialog)
        messageInt = 0
        clickListener = null
    }

    /**
     * 设置进度弹框显示文字
     */
    fun setMessageInt(@StringRes messageInt: Int) {
        this.messageInt = messageInt
    }
}

/**
 * 点击回调接口
 */
interface CommonLoadingDialogClickListener {
    /**
     * 点击取消
     */
    fun onCancel()
}
